contracts/contracts/strategies/balancer/BaseAuraStrategy.sol

```diff
@@ -78,8 +80,14 @@ abstract contract BaseAuraStrategy is BaseBalancerStrategy {
      * @param numBPTTokens Number of Balancer Pool Tokens (BPT) to withdraw
      */
     function _lpWithdraw(uint256 numBPTTokens) internal virtual override {
+        // Get all the strategy's BPTs in Aura
+        // maxRedeem is implemented as balanceOf(address) in Aura
+        uint256 maxBPTTokens = IERC4626(auraRewardPoolAddress).maxRedeem(
+            address(this)
+        );
+
         IRewardStaking(auraRewardPoolAddress).withdrawAndUnwrap(
-            numBPTTokens,
+            Math.min(numBPTTokens, maxBPTTokens),
```

Comment:
I do understand rationale for this, to make it more resilient, but I don't think it is an ok change. I think the caller of this function would expect for AURA to withdraw `numBPTTokens` or fail if it unable to withdraw all.